src/components/cc-logs-controller/cc-logs-controller.js

```diff
@@ -0,0 +1,504 @@
+  }
+
+  _onTimezoneChange ({ detail }) {
+    this.timezone = detail;
+    dispatchCustomEvent(this, 'optionChange', { name: 'timezone', value: this.timezone });
+  }
+
+  _onMetadataChange (e) {
+    const name = e.target.dataset.name;
+    const isHidden = !e.target.checked;
+    this.metadataDisplay = {
+      ...this.metadataDisplay,
+      [name]: {
+        ...this.metadataDisplay[name],
+        hidden: isHidden,
+      },
+    };
+    dispatchCustomEvent(this, 'optionChange', { name: 'metadata-display', value: this.metadataDisplay });
+  }
+
+  /* endregion */
+
+  /* region Private methods */
+
+  /**
+   * Resolves the metadata renderers according to the `metadataDisplay` and `metadataRenderers` properties.
+   *
+   * @param {{[key: string], MetadataRenderer}} metadataRenderers
+   * @param {{[key: string]: LogsMetadataDisplay}} metadataDisplay
+   * @return {[key: string], MetadataRenderer}
+   */
+  _resolveMetadataRenderers (metadataRenderers, metadataDisplay) {
+    return Object.fromEntries(
+      Object.entries(metadataRenderers).map(([name, renderer]) => {
+        const display = metadataDisplay[name];
+
+        return [
+          name,
+          display == null || !display.hidden ? renderer : { hidden: true },
+        ];
+      }),
+    );
+  }
+
+  /**
+   *
+   * @param {DateDisplay} dateDisplay
+   * @return {string}
+   */
+  _getDateDisplayLabel (dateDisplay) {
+    if (dateDisplay === 'none') {
+      return i18n('cc-logs-controller.date-display.none');
+    }
+    if (dateDisplay === 'datetime-iso') {
+      return i18n('cc-logs-controller.date-display.datetime-iso');
+    }
+    if (dateDisplay === 'time-iso') {
+      return i18n('cc-logs-controller.date-display.time-iso');
+    }
+    if (dateDisplay === 'datetime-short') {
+      return i18n('cc-logs-controller.date-display.datetime-short');
+    }
+    if (dateDisplay === 'time-short') {
+      return i18n('cc-logs-controller.date-display.time-short');
+    }
+  }
+
+  /**
+   *
+   * @param {Timezone} timezone
+   * @return {string}
+   */
+  _getTimezoneLabel (timezone) {
+    if (timezone === 'UTC') {
+      return i18n('cc-logs-controller.timezone.utc');
+    }
+    if (timezone === 'local') {
+      return i18n('cc-logs-controller.timezone.local');
+    }
+  }
+
+  /**
+   *
+   * @param {string} palette
+   * @return {string}
+   */
+  _getPaletteLabel (palette) {
+    if (palette === 'default') {
+      return i18n('cc-logs-controller.palette.default');
+    }
+    return palette;
+  }
+
+  /* endregion */
+
+  willUpdate (_changedProperties) {
+    if (_changedProperties.has('metadataRenderers') || _changedProperties.has('metadataDisplay')) {
+      this._resolvedMetadataRenderers = this._resolveMetadataRenderers(this.metadataRenderers, this.metadataDisplay);
+    }
+  }
+
+  render () {
+    return html`
+      <div class="header-slot">
+        <slot name="header"></slot>
+      </div>
```

Comment:
issue: can we add a story to show off this feature

or maybe we can remove this feature for now as we don't need it yet